tools/alive_parser.cpp

```diff
@@ -272,31 +274,55 @@ static Type& get_int_type(unsigned size) {
   return *int_types[size].get();
 }
 
-static Type& parse_type(bool optional = true) {
+static Type& get_pointer_type(unsigned address_space_number) {
+  if (address_space_number >= pointer_types.size())
+    pointer_types.resize(address_space_number + 1);
+
+  if (!pointer_types[address_space_number])
+    pointer_types[address_space_number] = make_unique<PtrType>(address_space_number);
+
+  return *pointer_types[address_space_number].get();
+}
+
+static Type& parse_type(bool can_fail = true) {
```

Comment:
typed pointers are being discontinued from LLVM. I would prefer to keep just `*` as the type (i.e., no `i32*` allowed).
The load/store operations receive the type for the operation instead as 1st argument.